rcl/src/rcl/node_options.c

```diff
@@ -50,6 +50,10 @@ rcl_node_options_copy(
     RCL_SET_ERROR_MSG("Attempted to copy options into itself");
     return RCL_RET_INVALID_ARGUMENT;
   }
+  if (NULL != options_out->arguments.impl) {
```

Comment:
Why do we need this? the same check will be done via rcl_arguments_copy. If we do not want to update any contents for options_out, we could move rcl_arguments_copy at the 1st.

Reply:
`rcl_arguments_copy` requires that the source impl is non-null. This helps with the case when the impl of the source argument is null.

The original motivation was that an uninitialized `rcl_node_options_t` when used as a destination for `rcl_node_options_copy` might cause a segfault in `rcl_node_options_fini` because options_out->arguments.impl was never set to NULL. The current behavior doesn't require the `arguments.impl` of the source in `rcl_node_options_copy` to be non-null. But if it is NULL, then the `arguments.impl` is not touched and it will remain an unitialized pointer.

This check just helps to ensure that the output node_options_t is valid when this function is called.

Reply:
understood.